types/sharedb/lib/client.d.ts

```diff
@@ -77,9 +77,9 @@ export class Connection extends ShareDB.TypedEmitter<ShareDB.ConnectionEventMap>
 
     ping(): void;
 }
-export type Doc<T = any> = ShareDB.Doc<T>;
+export class Doc<T = any> extends ShareDB.Doc<T> {}
```

Comment:
I'd suggest using the re-export syntax for these instead of exporting a new (non-existent) child class